### PR TITLE
CLOUD-901 variant-api 2.0.0

### PR DESCRIPTION
**Pull request by hpatel826**
# Description

After discussion with Vibin, library work has been paused due to my absence and so that we may achieve our DX release timeline goals. 

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## How Has This Been Tested?

- [X] unit tests
- [X] integration tests

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: charts/variant-cron/values.yaml [hunk 1]
```diff
@@ -25,12 +25,11 @@ podSecurityContext:
 
 securityContext:
   runAsNonRoot: true
-  runAsUser: 1000
   capabilities:
     drop:
     - ALL
   allowPrivilegeEscalation: false
-  readOnlyRootFilesystem: true
+  readOnlyRootFilesystem: false
```
Comment (hpatel826) [inline charts/variant-cron/values.yaml:32]: Lots of the data science containers will be writing temporary files so I relaxed this.

File: charts/variant-cron/values.yaml [hunk 2]
```diff
@@ -25,12 +25,11 @@ podSecurityContext:
 
 securityContext:
   runAsNonRoot: true
-  runAsUser: 1000
```
Comment (hpatel826) [inline charts/variant-cron/values.yaml:28]: `runAsNonRoot` above already requires that the container run as some user ID that is non-zero, so I removed this -- dont really care what user ID they are using

File: charts/variant-cron/values.yaml [hunk 3]
```diff
@@ -55,10 +54,10 @@ cronJob:
 awsSecrets: []
   # - name: eng-secret-in-aws
 
-secretVars: []
+secretVars: {}
```
Comment (hpatel826) [inline charts/variant-cron/values.yaml:57]: @LeeHughsVariant I noticed this and `configVars` are meant to be maps so this default should be `{}`

**Comment by rchengvariant** [comment]
@hpatel826 This looks good Tesh, can you squash and I'll approve. Also I think there's a Rebase and Merge thing you can do in github for PRs.

**Review by rchengvariant — APPROVED** [review]

**Pull request merged**